libs/dh/wholesale/feature-calculation-steps/src/lib/dh-wholesale-calculation-steps.component.ts

```diff
@@ -89,14 +96,61 @@ export class DhWholesaleCalculationStepsComponent {
     this.route.snapshot.params['gridAreaCode']
   );
 
+  ngOnInit() {
+    if (this.getCurrentStep()) {
+      this.openDrawer();
+    }
+  }
+
   getCurrentStep() {
-    // TODO: Is there a better way? This seems a little hacky.
-    return this.route.firstChild?.routeConfig?.path;
+    return this.route.firstChild?.snapshot.url?.[0]?.path;
+  }
+
+  openDrawer(commands?: unknown[]) {
+    if (commands) {
+      this.router.navigate(commands, { relativeTo: this.route });
+    }
+
+    // This is used to open the drawer when the user navigates to the page with a step in the url
+    this.isDrawerOpen = true;
+    // This is used to open the drawer when the user clicks on a step in the list
+    this.drawer?.open();
+
+    const step = commands ? String(commands[0]) : this.getCurrentStep();
+    const gln = this.route.firstChild?.snapshot.url?.[1]?.path;
+
+    this.getProcessStepResults(step, gln);
+  }
+
+  private getProcessStepResults(step?: string, gln = 'grid_area') {
+    combineLatest([this.batch$, this.gridArea$])
+      .pipe(
+        filter(([batch, gridArea]) => {
+          return !!batch && !!gridArea;
+        }),
+        first()
+      )
+      .subscribe(([batch, gridArea]) => {
+        if (batch && gridArea) {
+          this.store.getProcessStepResults({
+            batchId: batch.batchId,
+            gridAreaCode: gridArea.code,
+            timeSeriesType: this.getTimeSeriesType(step),
+            gln,
+          });
+        }
+      });
   }
 
-  openDrawer(step: number) {
-    this.router.navigate([step], { relativeTo: this.route });
-    this.drawer.open();
+  private getTimeSeriesType(step?: string) {
+    switch (step) {
+      case '1':
+        return TimeSeriesType.Production;
+      case '2':
+        return TimeSeriesType.NonProfiledConsumption;
+      default:
+        return TimeSeriesType.Production;
```

Comment:
question: should we add FlexConsumption to the list, just to be forwards-compatible?

Reply:
If we know it will be on step 3? otherwise, I think we should add it when the step which needs it is added.